### PR TITLE
Add bindings to support using chunked encoding.

### PR DESCRIPTION
**Pull request by JonathanHenson**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/test/java/software/amazon/awssdk/crt/test/HttpRequestResponseTest.java [hunk 1]
```diff
@@ -171,44 +178,65 @@ private boolean shouldRetry(TestHttpResponse response) {
         return false;
     }
 
-    public TestHttpResponse testRequest(String method, String endpoint, String path, String requestBody, int expectedStatus) throws Exception {
+    public TestHttpResponse testRequest(String method, String endpoint, String path, String requestBody, boolean useChunkedEncoding, int expectedStatus) throws Exception {
```
Comment (justinboswell) [inline src/test/java/software/amazon/awssdk/crt/test/HttpRequestResponseTest.java:181]: trivial: just add an overload signature for this function that is the old one, which calls this overload with `useChunkedEncoding` set to `false`, much smaller change.

**Review by bretambrose — APPROVED** [review]

File: src/main/java/software/amazon/awssdk/crt/http/HttpStream.java [hunk 2]
```diff
@@ -55,6 +63,57 @@ public void incrementWindow(int windowSize) {
...
+
+    /**
+     * Use only for Http 1.1 Chunked Encoding. At some later point we may adapt this interface for H2, but not yet. If
+     * using chunked encoding, remember, you'll need to send an empty chunk to signal the end of your stream. You must call
+     * activate() before using this function.
+     *
+     * @param chunkData chunk of data to send.
+     * @return completable future which will complete upon the data being flushed to the wire or an error occurring.
+     */
+    public CompletableFuture<Void> writeChunk(final byte[] chunkData) {
```
Comment (justinboswell) [inline src/main/java/software/amazon/awssdk/crt/http/HttpStream.java:99]: Should we make a `finishChunks()` or something that automates writing an empty chunk to finish the stream?

**Review by justinboswell — APPROVED** [review]
LGTM, feedback is optional

File: src/main/java/software/amazon/awssdk/crt/http/HttpStream.java [hunk 3]
```diff
@@ -55,6 +63,57 @@ public void incrementWindow(int windowSize) {
         }
     }
 
+    /**
+     * Use only for Http 1.1 Chunked Encoding. At some later point we may adapt this interface for H2, but not yet. If
```
Comment (rschmitt) [inline src/main/java/software/amazon/awssdk/crt/http/HttpStream.java:67]: How would h2 be any different?

File: src/main/java/software/amazon/awssdk/crt/http/HttpStream.java [hunk 4]
```diff
@@ -55,6 +63,57 @@ public void incrementWindow(int windowSize) {
...
+     * using chunked encoding, remember, you'll need to send an empty chunk to signal the end of your stream. You must call
```
Comment (rschmitt) [inline src/main/java/software/amazon/awssdk/crt/http/HttpStream.java:68]: I think this distinction will push unnecessary complexity into customer code. If you use a `Content-Length`, can you publish an empty chunk at the end of the stream?

File: src/main/java/software/amazon/awssdk/crt/http/HttpStream.java (hunk 3 above)
Comment (JonathanHenson) [reply src/main/java/software/amazon/awssdk/crt/http/HttpStream.java:67]: If you were sending something over h2 for a use case you'd normally use chunked encoding for in http 1.1, the API should be identical. But, I don't think it's plumbed together in aws-c-http at the moment. When it is, I'll connect the pieces here and change the comment.

File: src/test/java/software/amazon/awssdk/crt/test/HttpRequestResponseTest.java (hunk 1 above)
Comment (JonathanHenson) [reply src/test/java/software/amazon/awssdk/crt/test/HttpRequestResponseTest.java:181]: Seems to just make it less clear and explicit to me, but doesn't shave off the majority of the code I needed to change.

**Review by justinboswell — APPROVED** [review]

**Pull request merged**